### ISSUE TITLE
Microtile closure

### ISSUE THREAD

**Issue by markriegler** [tataratat/splinepy#460]
Some microtiles have a faulty closure behaviour. In particular the following tiles:

[ ] HollowOctagonExtrude
[ ] InverseCross3D

The HollowOctagonExtrude does not seem to have a closure at all. And the InverseCross3D does not fill the whole deformation function.

**Comment by markriegler** [comment]
I will leave the InverseCross3D as it is, since the closure behaviour is very different to the others, in particular:

- The closure tile does not fill the whole unit cube
- The number of patches is different in the closing tile compared to a regular tile

![InverseCross-Closure](https://github.com/user-attachments/assets/fc5d703a-f9ff-4c74-a6d9-0f6f2b0da314)

**Comment by markriegler** [comment]
Also, the InverseCross3D should be the inverse domain of the Cross3D. Meaning, that the generated microstructure is the inverse domain of a microstructure with the Cross3D microtile and a closure. Hence, the closure tile shouldn't fill the whole unit cube.

**Comment by jzwar** [comment]
Yes, that was the whole point of the inverse tiles. They are supposed to fill the unit cube entirely by being everywhere the positive tiles (cross tiles) are not. They cannot be on the closing face! That would defeat the whole purpose of the watertightness. @danielwolff1 has already performed some fluid calculations through that grid. It's important that they have the same input parameters. To check they behave correctly, you could calculate the volume of the tiles for different parameters (very accurately) and ensure that the sum of the inverse and positive tiles always equals the volume of the unit cube. You could also extract faces from the tiles for various parameters and check the interfaces correspond one-to-one. All surfaces must be duplicates. We would also need to check the closing faces here!

It's a different story for the hollow cube/ellipsoid. They don't require closed faces because they are enclosed by every grid tile. The void spaces are entirely within the structure. So there is no need for closing faces here. These were originally designed for a test case that we had with a project partner. Please ask Steffi or me if you need more details.
